net/libfab/libfab.c

```diff
@@ -2778,10 +2779,8 @@ static int libfab_domain_params_get(struct m0_fab__ndom *fab_ndom)
 		fab_ndom->fnd_seg_nr = FAB_TCP_SOCK_IOV_MAX;
 		fab_ndom->fnd_seg_size = FAB_TCP_SOCK_MAX_BULK_SEG_SIZE;
 	}
-
 	hints->fabric_attr->prov_name = NULL;
 	fi_freeinfo(hints);
-	fi_freeinfo(fi);
```

Comment:
fi_freeinfo() works similar to m0_free(). i.e if 'fi' is NULL, then it would return without doing anything.
Hence call to fi_freeinfo() here was safe.
But I agree that calling fi_freeinfo only when fi is allocated is a better way to free the list.

Reply:
It wasn't NULL: fi variable is not initialised, so it remains garbage after failed fi_getinfo(), I observed this in UT.

Reply:
As per the man page of fi_getinfo(), 'fi' will be set to NULL if it fails.
```
int fi_getinfo(int version, const char *node, const char *service,
uint64_t flags, const struct fi_info *hints, struct fi_info **info);

fi_getinfo returns information about available fabric services for reaching specified node or service, subject to any provided hints. Callers may specify NULL for node, service, and hints in order to retrieve information about what providers are available and their optimal usage models. If no matching fabric information is available, info will be set to NULL and the call will return -FI_ENODATA.
```

Could you please let me know in which UT this failure was seen.